clusterloader2/framework/util/util.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+func CreateNamespaces(f *framework.Framework, basename string, namespaceCount int) ([]*apiv1.Namespace, error) {
+	namespaces := []*apiv1.Namespace{}
+	for i := 0; i < namespaceCount; i++ {
+		namespace, err := f.CreateNamespace(fmt.Sprintf("%v-%d", basename, i), nil)
+		if err != nil {
+			return []*apiv1.Namespace{}, err
+		}
+		namespaces = append(namespaces, namespace)
+	}
+	return namespaces, nil
+}
+
+func CreateObject(dynamicClient dynamic.Interface, namespace string, gvk schema.GroupVersionKind, name string, obj *unstructured.Unstructured) error {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	obj.SetName(name)
+	_, err := dynamicClient.Resource(gvr).Namespace(namespace).Create(obj)
+	return err
+}
+
+func DeleteObject(dynamicClient dynamic.Interface, namespace string, gvk schema.GroupVersionKind, name string) error {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	return dynamicClient.Resource(gvr).Namespace(namespace).Delete(name, nil)
+}
+
+func GetObject(dynamicClient dynamic.Interface, namespace string, gvk schema.GroupVersionKind, name string) (*unstructured.Unstructured, error) {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	return dynamicClient.Resource(gvr).Namespace(namespace).Get(name, metav1.GetOptions{
+		IncludeUninitialized: true,
+	})
+}
+
+// TODO(krzysied): Add update object function
```

Comment:
Can we do that together - how is that different from create and delete?

Reply:
If I'm not wrong, kubernetes has different implementation of update. In most cases Delete -> Create is used when this default method fails.

I will add Delete->Create here, but at some point I want to update this code, so it uses correct update call.